ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -106,22 +107,39 @@ public void close() {
       final QueryId queryId,
       final PullQueryQueue pullQueryQueue
   ) {
-    final List<KsqlPartitionLocation> locations = pullPhysicalPlan.getMaterialization().locator()
+    final List<KsqlPartitionLocation> allLocations = pullPhysicalPlan.getMaterialization().locator()
         .locate(
             pullPhysicalPlan.getKeys(),
             routingOptions,
             routingFilterFactory
     );
 
-    final boolean anyPartitionsEmpty = locations.stream()
-        .anyMatch(location -> location.getNodes().isEmpty());
-    if (anyPartitionsEmpty) {
-      LOG.debug("Unable to execute pull query: {}. All nodes are dead or exceed max allowed lag.",
-                statement.getStatementText());
-      throw new MaterializationException("Unable to execute pull query. "
-          + "All nodes are dead or exceed max allowed lag.");
```

Comment:
Just making sure: the message `or exceed max allowed lag` is actually incorrect and hence we exclude it in the PR, right?

Reply:
it could've been correct, but it also may not have been. the new error message will say whether the hosts were dead, whether the lag was exceeded, or something else entirely happened (no hosts were found at all for that partition)